grpc/naming.go

```diff
@@ -84,10 +100,15 @@ func (w *watcher) Next() ([]*naming.Update, error) {
 	}
 
 	var lastErr error
-	for i := 0; i < MaximumConsecutiveErrors; i++ {
+	consecutiveErrors := 0
+	for {
 		freshTargets, err := w.resolver.Lookup(w.domainName)
 		if err != nil {
 			lastErr = err
+			consecutiveErrors += 1
+			if consecutiveErrors >= w.maxConsecutiveErrors && w.maxConsecutiveErrors != 0 {
```

Comment:
do you not need a special case for detecting the `-1` for `macConsecutiveErrors`?